### PR TITLE
Kafka on EKS - EKS Bluprints V5 Mitigation

### PR DESCRIPTION
**Pull request by AlyIbrahim**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Updating Kafka-on-EKS patterns by:
- Removing EKS Blueprints dependencies
- Updating Strimzi Operator to v0.34
- Update Kafka to v3.4

### Motivation

EKS Blueprints v5 upgrade 

### More

- [X] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [X] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [X] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: streaming/kafka/addons.tf [hunk 1]
```diff
@@ -1,152 +1,97 @@
...
+  tags = local.tags
+}
+
+#---------------------------------------------------------------
+# Kubernetes Add-ons
+#---------------------------------------------------------------
+resource "aws_eks_addon" "vpc_cni" {
+  cluster_name = module.eks.cluster_name
+  addon_name   = "vpc-cni"
+  service_account_role_arn = module.vpc_cni_irsa.iam_role_arn
```
Comment (vara-bonthu) [inline streaming/kafka/addons.tf:43]: add `preserve=true` to ensure VPC CNI is not fully deleted before the EKS Cluster. VPC CNI sometimes required to cleanup the existing add-ons

File: streaming/kafka/addons.tf [hunk 2]
```diff
@@ -1,152 +1,97 @@
...
+  depends_on = [module.eks.eks_managed_node_groups]
+}
+
+resource "aws_eks_addon" "coredns" {
+  cluster_name = module.eks.cluster_name
+  addon_name   = "coredns"
```
Comment (vara-bonthu) [inline streaming/kafka/addons.tf:49]: add `preserve=true`

File: streaming/kafka/addons.tf [hunk 3]
```diff
@@ -1,152 +1,97 @@
...
+  depends_on = [module.eks.eks_managed_node_groups]
+}
+
+resource "aws_eks_addon" "kube_proxy" {
+  cluster_name = module.eks.cluster_name
+  addon_name   = "kube-proxy"
```
Comment (vara-bonthu) [inline streaming/kafka/addons.tf:55]: add `preserve=true`

File: streaming/kafka/addons.tf [hunk 4]
```diff
@@ -1,152 +1,97 @@
...
+resource "aws_eks_addon" "aws_ebs_csi_driver" {
+  cluster_name = module.eks.cluster_name
+  addon_name   = "aws-ebs-csi-driver"
+  service_account_role_arn = module.ebs_csi_driver_irsa.iam_role_arn
+  depends_on = [module.eks.eks_managed_node_groups]
+}
+
+#---------------------------------------------------------------
+# Cluster Autoscaler
+#---------------------------------------------------------------
```
Comment (vara-bonthu) [inline streaming/kafka/addons.tf:69]: nit:remove extra line

File: streaming/kafka/addons.tf [hunk 5]
```diff
@@ -169,85 +114,115 @@ provisioner: ebs.csi.aws.com
 reclaimPolicy: Delete
 volumeBindingMode: WaitForFirstConsumer
 YAML
+}
```
Comment (vara-bonthu) [inline streaming/kafka/addons.tf:117]: Add a dependency to on EKS module for this storage class resource

File: streaming/kafka/addons.tf [hunk 6]
```diff
@@ -169,85 +114,115 @@ provisioner: ebs.csi.aws.com
...
+    wait             = true
+    values           = [templatefile("${path.module}/helm-values/prom-grafana-values.yaml", {})]
+    depends_on = [aws_eks_addon.vpc_cni]
+  }
 
-  depends_on = [helm_release.grafana_operator]
-}
+  resource "kubectl_manifest" "podmonitor_cluster_operator_metrics" {
+  yaml_body = file("./monitoring-manifests/podmonitor-cluster-operator-metrics.yml")
+  depends_on = [kubectl_manifest.kafka_namespace]
```
Comment (vara-bonthu) [inline streaming/kafka/addons.tf:197]: nit: formatting. add two spaces. Pre-commit didn't pickup thisone?
same for all the resources below

File: streaming/kafka/main.tf [hunk 7]
```diff
@@ -36,105 +48,139 @@ module "eks_blueprints" {
...
+    # }
+
+    # ingress_fsx2 = {
+    #   description = "Allows Lustre traffic between Lustre clients"
+    #   cidr_blocks = module.vpc.private_subnets_cidr_blocks
+    #   from_port   = 988
+    #   to_port     = 988
+    #   protocol    = "tcp"
+    #   type        = "ingress"
+    # }
```
Comment (vara-bonthu) [inline streaming/kafka/main.tf:68]: remove the commented lines

**Review by vara-bonthu — COMMENTED** [review]
Thanks @AlyIbrahim ! PR Looks good. few minor comment. 
In addition.
1/ Please add install and cleanup scripts like other blueprints
2/ Raise an issue to create a Architecture diagram for this pattern

Thanks 👍🏼

File: streaming/kafka/addons.tf (hunk 1 above)
Comment (AlyIbrahim) [reply streaming/kafka/addons.tf:43]: Done

File: streaming/kafka/addons.tf (hunk 2 above)
Comment (AlyIbrahim) [reply streaming/kafka/addons.tf:50]: Done

File: streaming/kafka/addons.tf (hunk 3 above)
Comment (AlyIbrahim) [reply streaming/kafka/addons.tf:57]: Done

File: streaming/kafka/addons.tf (hunk 4 above)
Comment (AlyIbrahim) [reply streaming/kafka/addons.tf]: Done

File: streaming/kafka/addons.tf (hunk 6 above)
Comment (AlyIbrahim) [reply streaming/kafka/addons.tf]: Done

File: streaming/kafka/main.tf (hunk 7 above)
Comment (AlyIbrahim) [reply streaming/kafka/main.tf]: Done

**Review by vara-bonthu — APPROVED** [review]
Thanks for the PR @AlyIbrahim ! LGTM 👍🏼

**Review by vara-bonthu — APPROVED** [review]
LGTM 👍🏼 Thanks for the PR 🚀

**Pull request merged**